libraries/Zigbee/examples/Zigbee_Pressure_Flow_Sensor/Zigbee_Pressure_Flow_Sensor.ino

```diff
@@ -0,0 +1,134 @@
+    float flow_value = temperatureRead();
+    uint16_t pressure_value = (uint16_t)temperatureRead()*100; //*100 for demonstration so the value is in 1-3hPa
+    Serial.printf("Updating flow sensor value to %.2f\r\n", flow_value);
+    zbFlowSensor.setFlow(flow_value);
+    Serial.printf("Updating pressure sensor value to %d\r\n", pressure_value);
+    zbPressureSensor.setPressure(pressure_value);
+    delay(1000);
+  }
+}
+
+/********************* Arduino functions **************************/
+void setup() {
+  Serial.begin(115200);
+
+  // Init button switch
+  pinMode(button, INPUT_PULLUP);
+
+  // Optional: set Zigbee device name and model
+  zbFlowSensor.setManufacturerAndModel("Espressif", "ZigbeeFlowSensor");
+
+  // Set minimum and maximum temperature measurement value (10-50°C is default range for chip temperature measurement)
+  zbFlowSensor.setMinMaxValue(0, 100);
+
+  // Set tolerance for temperature measurement in °C (lowest possible value is 0.01°C)
+  zbFlowSensor.setTolerance(1);
+
+  // Optional: set Zigbee device name and model
+  zbPressureSensor.setManufacturerAndModel("Espressif", "ZigbeePressureSensor");
+
+  // Set minimum and maximum temperature measurement value (10-50°C is default range for chip temperature measurement)
+  zbPressureSensor.setMinMaxValue(0, 30);
+
+  // Set tolerance for temperature measurement in °C (lowest possible value is 0.01°C)
+  zbPressureSensor.setTolerance(1);
+
+  // Add endpoints to Zigbee Core
+  Zigbee.addEndpoint(&zbFlowSensor);
+  Zigbee.addEndpoint(&zbPressureSensor);
+
+  Serial.println("Starting Zigbee...");
+  // When all EPs are registered, start Zigbee in End Device mode
+  if (!Zigbee.begin()) {
+    Serial.println("Zigbee failed to start!");
+    Serial.println("Rebooting...");
+    ESP.restart();
+  } else {
+    Serial.println("Zigbee started successfully!");
+  }
+  Serial.println("Connecting to network");
+  while (!Zigbee.connected()) {
+    Serial.print(".");
+    delay(100);
+  }
+  Serial.println();
+
+  // Start Flow and Pressure sensor reading task
+  xTaskCreate(sensors_reading, "flow_pressure_sensors_read", 2048, NULL, 10, NULL);
+
+  // Set reporting interval for flow and pressure measurement in seconds, must be called after Zigbee.begin()
+  // min_interval and max_interval in seconds, delta (pressure change in Pa, flow change in 0,1 m3/h)
+  // if min = 1 and max = 0, reporting is sent only when temperature changes by delta
+  // if min = 0 and max = 10, reporting is sent every 10 seconds or temperature changes by delta
+  // if min = 0, max = 10 and delta = 0, reporting is sent every 10 seconds regardless of delta change
+  zbFlowSensor.setReporting(0, 30, 1);
+  zbPressureSensor.setReporting(0, 30, 1);
+}
+
+void loop() {
+  // Checking button for factory reset
+  if (digitalRead(button) == LOW) {  // Push button pressed
+    // Key debounce handling
+    delay(100);
+    int startTime = millis();
+    while (digitalRead(button) == LOW) {
```

Comment:
`while (digitalRead(button) == LOW)` will block `loop()` for the time the button is kept pressed or when 3 seconds has passed.

Maybe an impementation that uses time stamp - non blocking - would be better.

Reply:
That is OK-ish here. It will not wait more than the reset time to do a factory reset and shorter reset will trigger report of values

Reply:
The sketch uses a Task to publish the sensor data.
It could be done in the `loop()` and remove the Task.
But for that, the change I suggested is necessary.

I think that it would make it more Arduinish.

Reply:
Done